src/LibraryViewExtensionMSWebBrowser/LibraryViewController.cs

```diff
@@ -0,0 +1,581 @@
+                }
+                //Create the node of given item name
+                var cmd = new DynamoModel.CreateNodeCommand(Guid.NewGuid().ToString(), nodeName, -1, -1, true, false);
+                commandExecutive.ExecuteCommand(cmd, Guid.NewGuid().ToString(), LibraryViewExtensionMSWebBrowser.ExtensionName);
+                LogEventsToInstrumentation(CreateNodeInstrumentationString, nodeName);
+
+                this.disableObserver = false;
+            }));
+        }
+
+        /// Call this method to import a zero touch library. It will prompt
+        /// user to select the zero touch dll.
+        /// </summary>
+        public void ImportLibrary()
+        {
+            dynamoWindow.Dispatcher.BeginInvoke(new Action(() =>
+                dynamoViewModel.ImportLibraryCommand.Execute(null)
+            ));
+        }
+        /// <summary>
+        /// This function logs events to instrumentation if it matches a set of known events
+        /// </summary>
+        /// <param name="eventName">Event Name that gets logged to instrumentation</param>
+        /// <param name="data"> Data that gets logged to instrumentation </param>
+        public void LogEventsToInstrumentation(string eventName, string data)
+        {
+            if (eventName == "Search" || eventName == "Filter-Categories" || eventName == "Search-NodeAdded")
+            {
+                Analytics.LogPiiInfo(eventName, data);
+            }
+        }
+        internal void RefreshLibraryView(WebBrowser browser)
+        {
+            dynamoWindow.Dispatcher.BeginInvoke(
+            new Action(() =>
+            {
+
+                var ext1 = string.Empty;
+                var ext2 = string.Empty;
+                //read the full loadedTypes and LayoutSpec json strings.
+                var reader = new StreamReader(nodeProvider.GetResource(null, out ext1));
+                var loadedTypes = reader.ReadToEnd();
+
+                var reader2 = new StreamReader(layoutProvider.GetResource(null, out ext2));
+                var layoutSpec = reader2.ReadToEnd();
+
+                try
+                {
+                    browser.InvokeScript
+                         ("refreshLibViewFromData", loadedTypes, layoutSpec);
+                }
+                catch (Exception e)
+                {
+                    this.dynamoViewModel.Model.Logger.Log(e);
+                }
+                reader.Dispose();
+                reader2.Dispose();
+
+            }));
+
+        }
+
+        #endregion
+
+        private string ReplaceUrlWithBase64Image(string html, string minifiedURL, bool magicreplace = true)
+        {
+            var ext = string.Empty;
+            // this depends on librariejs minification producing the same results - 
+            // longterm this is fragile. We should intercept these requests and handle them instead.
+            if (magicreplace)
+            {
+                minifiedURL = $"n.p+\"{minifiedURL}\"";
```

Comment:
What does this do?

Reply:
there are strings in the minified librarie.js file that look like this `n.p\resoruces\some\path\toAfont.font` - the n.p is some artifact of the minification - to replace these with embedded base64 versions of those resources we need to search for the whole string including the n.p part.